--- CMakeLists.txt
+++ CMakeLists.txt
@@ -54,10 +54,11 @@
   LIBRARY_OUTPUT_DIRECTORY ${CATKIN_DEVEL_PREFIX}/${PYTHON_INSTALL_DIR}/roslz4)
 if (NOT WIN32)
   set_target_properties(roslz4_py PROPERTIES OUTPUT_NAME roslz4 PREFIX "_" SUFFIX ".so")
+  set_target_properties(roslz4_py PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
 else()
   set_target_properties(roslz4_py PROPERTIES OUTPUT_NAME _roslz4 SUFFIX ".pyd")
 endif()
-target_link_libraries(roslz4_py roslz4 ${catkin_LIBRARIES} ${PYTHON_LIBRARIES})
+target_link_libraries(roslz4_py roslz4 ${catkin_LIBRARIES})
 endif()
 
 # Install library

